server/src/services/apps.service.ts

```diff
@@ -294,6 +294,14 @@ export class AppsService {
       where: { id: versionFromId },
     });
 
+    const versionNameExists = await this.appVersionsRepository.findOne({
+      where: { name: versionName, appId: app.id },
+    });
+
+    if (versionNameExists) {
+      throw new BadRequestException('Version name already exists.');
+    }
+
     let appVersion: AppVersion;
     await getManager().transaction(async (manager) => {
```

Comment:
```suggestion
    try {
      await getManager().transaction(async (manager) => {
        appVersion = await manager.save(
          AppVersion,
          manager.create(AppVersion, {
            name: versionName,
            app,
            definition: versionFrom?.definition,
            createdAt: new Date(),
            updatedAt: new Date(),
          })
        );
        await this.setupDataSourcesAndQueriesForVersion(manager, appVersion, versionFrom);
      });
    } catch (error) {
      if (error.code === '23505' && error.constraint === 'name_app_id_app_versions_unique') {
        throw new BadRequestException('Version name already exists.');
      }
      throw new Error();
    }
```
Is it feasible? we could save an extra query
@akshaysasidrn

Reply:
This is feasible, but let's make changes to handle db exceptions separately on a different PR. We can handle this throughout the app to return appropriate responses gracefully. @shah21 is looking into this already.